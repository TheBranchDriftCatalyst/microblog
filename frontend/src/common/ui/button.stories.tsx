import { fn, userEvent, within } from "@storybook/test";
import { Button } from "./button";

export default {
  title: "UI/Button",
  component: Button,
  tags: ["autodocs"],
  argTypes: {
    variant: {
      options: ["default", "destructive", "outline", "secondary", "link"],
      control: { type: "radio" },
      defaultValue: "default",
    },
    size: {
      options: ["default", "xxxs", "xxs", "xs", "sm", "lg", "icon"],
      control: { type: "radio" },
      defaultValue: "default",
    },
    onClick: { action: "clicked" },
  },
  args: {
    onClick: fn(),
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const button = canvas.getByRole("button");
    await userEvent.click(button);
  },
};

export const Default = {
  args: {
    children: "Button",
  },
};

export const Destructive = {
  args: {
    variant: "destructive",
    children: "Destructive",
  },
};

export const Outline = {
  args: {
    variant: "outline",
    children: "Outline",
  },
};

export const Secondary = {
  args: {
    variant: "secondary",
    children: "Secondary",
  },
};

export const Ghost = {
  args: {
    variant: "ghost",
    children: "Ghost",
  },
};

export const Link = {
  args: {
    variant: "link",
    children: "Link",
  },
};

// You can also add stories for different sizes if needed
export const SmallSize = {
  args: {
    size: "sm",
    children: "Small Button",
  },
};

export const LargeSize = {
  args: {
    size: "lg",
    children: "Large Button",
  },
};

export const IconSize = {
  args: {
    size: "icon",
    children: "Icon Button",
  },
};
