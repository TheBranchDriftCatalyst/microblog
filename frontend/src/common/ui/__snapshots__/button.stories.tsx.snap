// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Button Default smoke-test 1`] = `
<button class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2">
  Button
</button>
`;

exports[`UI/Button Destructive smoke-test 1`] = `
<button class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-destructive text-destructive-foreground hover:bg-destructive/90 h-10 px-4 py-2">
  Destructive
</button>
`;

exports[`UI/Button Ghost smoke-test 1`] = `
<button class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2">
  Ghost
</button>
`;

exports[`UI/Button IconSize smoke-test 1`] = `
<button class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 w-10 rounded">
  Icon Button
</button>
`;

exports[`UI/Button LargeSize smoke-test 1`] = `
<button class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-11 rounded-md px-8">
  Large Button
</button>
`;

exports[`UI/Button Link smoke-test 1`] = `
<button class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-10 px-4 py-2">
  Link
</button>
`;

exports[`UI/Button Outline smoke-test 1`] = `
<button class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2">
  Outline
</button>
`;

exports[`UI/Button Secondary smoke-test 1`] = `
<button class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2">
  Secondary
</button>
`;

exports[`UI/Button SmallSize smoke-test 1`] = `
<button class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3">
  Small Button
</button>
`;
